jenkins/Jenkinsfile_noms

```diff
@@ -0,0 +1,88 @@
+pipeline {
+  agent {
+        node {
+            label 'team:makerspace'
+        }
+    }
+    environment {
+     GA_AUTH_FILE = credentials('GOOGLE_ANALYTICS_SERVICE_JSON')   
+     GA_CLIENTID_FILE = credentials('GOOGLE_ANALYTICS_CLIENTID_JSON')
+    }
+    parameters {
+        gitParameter name: 'BRANCH_TAG',
+                     type: 'PT_BRANCH_TAG',
+                     defaultValue: 'v0.0.6'
+        choice(choices: ['test', 'prod'], description: 'Tier to deploy files to', name: 'TIER')
+    }
+    triggers {
+        parameterizedCron('''
+          # leave spaces where you want them around the parameters. They'll be trimmed.
+          H(45-59) 7 * * * %BRANCH_TAG=v0.0.6; TIER=test
+        ''') 
+    }
+    options {
+      buildDiscarder(logRotator(numToKeepStr: '5'))
+    }
+  stages {
+    stage('Clean Workspace') {
+      steps{
+        cleanWs()
+      }
+    }
+    stage('Checkout repo') {
+      steps {
+        sh 'wget -O DOIRootCA2.cer http://sslhelp.doi.net/docs/DOIRootCA2.cer'
+        checkout([$class: 'GitSCM',
+                          branches: [[name: "${params.BRANCH_TAG}"]],
+                          doGenerateSubmoduleConfigurations: false,
+                          extensions: [],
+                          gitTool: 'Default',
+                          submoduleCfg: [],
+                          userRemoteConfigs: [[url: 'https://github.com/usgs-makerspace/analytics_pipeline']]
+                        ])
+      }
+    }
+    stage('pull Google Analytics data') {
+      agent {
+        docker {
+          image 'code.chs.usgs.gov:5001/wma/iidd/analytics_pipeline:R-latest'
+          registryUrl 'https://code.chs.usgs.gov:5001/wma/iidd/analytics_pipeline'
+          registryCredentialsId 'jenkins_ci_access_token'
+          alwaysPull true
+          reuseNode true
+          label 'team:makerspace'
+        } 
+      }
+      steps {
+        retry(2){
+          sh 'Rscript -e "source(\'R/pull_NOMS_data.R\')"'
+        }
+      }
+    }
+    stage('push to S3') {
+      steps { 
+        sh '''
+          # --delete flag important so old files don't hang around in S3 if file names change
```

Comment:
is this actually not what I want if I want to push a bunch of files to s3 that I ran off-jenkins? wondering if the sync would copy all those files back onto jenkins which would be weird and/or delete them in s3 if they aren't in the jenkins workspace? maybe i should just use cp?

Reply:
We definitely don't want the `--delete` option on Jenkins or off—this pull is adding new files each day instead of overwriting like the dashboard job does.  I think either sync without --delete or cp would work, sync just is recursive automatically is the main difference I think.